### ISSUE TITLE
a11y Unable to check/uncheck Checkbox using keyboard

### ISSUE THREAD

**Issue by natmasse** [IBM/carbon-components-react#1782]
a11y Unable to check/uncheck Checkbox using keyboard

## Detailed description
For accessibility, a user should be able to tab to a checkbox and use the keyboard to check/uncheck the checkbox. 

> Describe in detail the issue you're having.
The checkbox is not operable with the keyboard.

> Is this issue related to a specific component?
Checkbox

> What did you expect to happen? What happened instead? What would you like to see changed?
Expect the checkbox to toggle between check/uncheck  when pressing "enter" key

> What browser are you working in?
Chrome

> What version of the Carbon Design System are you using?
carbon-components-react v6.81.0

> What offering/product do you work on? Any pressing ship or release dates we should be aware of?
IBM cloud console

## Steps to reproduce the issue

1. Tab to checkbox
2. Press enter key => nothing happens

**Comment by asudoh** [comment]
Hi 👋 checkbox works with space key instead of enter key, including native ones. Wondering if you have tried it? Thanks!

**Comment by natmasse** [comment]
Space is working. Thanks.
